### ISSUE TITLE
pg_default_acl is not populated by `ALTER DEFAULT PRIVILEGES`

### ISSUE THREAD

**Issue by kylepl** [cockroachdb/cockroach#114032]
**Describe the problem**

Please describe the issue you observed, and any steps we can take to reproduce it:

**To Reproduce**

While logged-in as a superuser, with another user "foo" created that you want to grant rights to, run:

```
ALTER DEFAULT PRIVILEGES IN SCHEMA "public" GRANT UPDATE,SELECT,DELETE,INSERT ON TABLES TO "foo";
SELECT * FROM pg_default_acl;
```

This returns:
```
 oid | defaclrole | defaclnamespace | defaclobjtype | defaclacl 
-----+------------+-----------------+---------------+-----------
(0 rows)
```

**Expected behavior**
It should return a row for the new granted default ACL.

**Environment:**
 - CockroachDB version - v23.1.11
 - Server OS: Linux
 - Client app - psql

**Additional context**
Note that I found a [TODO](https://github.com/cockroachdb/cockroach/blob/dfbc963e79751f0edf63b9e178a098b5d37ba0dd/pkg/sql/pg_catalog.go#L1524) that seems relevant in `pgCatalogDefaultACLTable` that was potentially missed after https://github.com/cockroachdb/cockroach/issues/67376 was finished. @RichardJCai as FYI.

This came up trying to use the [Terraform postgres provider](https://github.com/cyrilgdn/terraform-provider-postgresql/tree/a961e75c59b360f901020c612deef1e1da8ecfea). It needs to be able to query the current state of default ACLs for [postgresql_default_privileges](https://registry.terraform.io/providers/cyrilgdn/postgresql/latest/docs/resources/postgresql_default_privileges).

---
### To resolve:

Take a look at https://github.com/cockroachdb/cockroach/blob/dfbc963e79751f0edf63b9e178a098b5d37ba0dd/pkg/sql/pg_catalog.go#L1553

In addition to getting the DefaultPrivilegeDescriptor for the database, we should also get the DefaultPrivilegeDescriptor for each schema in that database. Use the `forEachSchema` helper function to do this.

We likely will need to modify the signature of the [helper function](https://github.com/cockroachdb/cockroach/blob/dfbc963e79751f0edf63b9e178a098b5d37ba0dd/pkg/sql/pg_catalog.go#L1375) so that it accepts `schemaID` as a parameter.

---

Jira issue: CRDB-33309

Epic CRDB-35306

**Comment by rafiss** [comment]
Thanks for this report! You are right that the TODO you found explains why this information is missing. The original author isn't working on this now, but we will work on fixing this.

**Comment by kylepl** [comment]
Thanks for the fix! Should I assume this will first be available in v24.2?

**Comment by rafiss** [comment]
That's right, this will be in 24.2!
